crates/rpc-types-engine/src/payload/mod.rs

```diff
@@ -3,19 +3,201 @@
 pub mod v3;
 pub mod v4;
 
-use crate::OpExecutionPayloadV4;
-use alloy_rpc_types_engine::{ExecutionPayloadV2, ExecutionPayloadV3};
+use crate::OpExecutionPayloadSidecar;
+use alloy_consensus::{Block, BlockHeader, Transaction, EMPTY_ROOT_HASH};
+use alloy_eips::{BlockNumHash, Decodable2718, Encodable2718};
+use alloy_primitives::B256;
+use alloy_rpc_types_engine::{
+    ExecutionPayloadV1, ExecutionPayloadV2, ExecutionPayloadV3, PayloadError,
+};
+use derive_more::From;
 
-/// An execution payload, which can be either [`ExecutionPayloadV2`], [`ExecutionPayloadV3`], or
-/// [`OpExecutionPayloadV4`].
-#[derive(Clone, Debug, PartialEq, Eq)]
-#[cfg_attr(feature = "serde", derive(serde::Serialize))]
+/// An execution payload, which can be either [`ExecutionPayloadV2`] and [`ExecutionPayloadV3`].
+#[derive(Clone, Debug, PartialEq, Eq, From)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[cfg_attr(feature = "serde", serde(untagged))]
 pub enum OpExecutionPayload {
     /// V2 payload
     V2(ExecutionPayloadV2),
     /// V3 payload
     V3(ExecutionPayloadV3),
-    /// V4 payload
-    V4(OpExecutionPayloadV4),
```

Comment:
why is this removed?

Reply:
ye

Reply:
see https://specs.optimism.io/protocol/isthmus/exec-engine.html#update-to-executabledata

Reply:
still unclear what the parameters are, from reading the spec I'd assume that the withdrawals root is an additional attribute of the payload param and not a standalone parameter?

Reply:
yes, we know, we are working on it